src/main/java/games/strategy/engine/framework/map/download/DownloadFileParser.java

```diff
@@ -26,38 +27,31 @@ private DownloadFileParser() {}
   }
 
   public static List<DownloadFileDescription> parse(final InputStream is) {
-    @SuppressWarnings("unchecked")
-    final List<Map<String, String>> yamlData = new Yaml().loadAs(is, List.class);
+    final JSONArray yamlData = new JSONArray(new Yaml().loadAs(is, List.class));
 
     final List<DownloadFileDescription> rVal = new ArrayList<>();
-    for (final Map<String, String> yaml : yamlData) {
-      final String url = yaml.get(Tags.url.toString());
-      final String description = yaml.get(Tags.description.toString());
-      final String mapName = yaml.get(Tags.mapName.toString());
-
-      Version version = null;
-      final Object versionObject = yaml.get(Tags.version.toString());
-      if (versionObject != null) {
-        version = new Version(String.valueOf(versionObject));
-      }
-
+    for (int i = 0; i < yamlData.length(); i++) {
+      final JSONObject yaml = yamlData.getJSONObject(i);
+      final String url = yaml.getString(Tags.url.toString());
```

Comment:
The get methods throw an exception whenever the searched key is not present, this is intentional, as some sort of "fail-fast" check, because HashMap returns null, which often leads to NullPointerExceptions